ckanext/switzerland/logic.py

```diff
@@ -534,3 +537,34 @@ def ogdch_user_create(context, data_dict):
             .format(user['email'], user['name'], exception))
 
     return user
+
+
+def ogdch_subscribe_manage(context, data_dict):
+    '''Request a code for managing existing subscriptions. Causes a email to be
+    sent, containing a manage link.
+    :returns: list of dictionaries
+    '''
+    
+    data_dict['email'] = authenticate_with_code(data_dict['code'])
```

Comment:
This line should be in a try except clause: the function `authenticate_with_code` raises a ValueError, if it fails as you can see here: https://github.com/opendata-swiss/ckanext-subscribe/blob/acb56d5b5cf7bd5bace10611e2ffb0d31e859226/ckanext/subscribe/email_auth.py#L121 

This is the error you need to catch: otherwise CKAN will crash on it. And then you want to return a ValidationError to the API since the code did not validate. This error has a special form. So your try catch clause would be something like this; 

```
try: 
    data_dict['email'] = authenticate_with_code(data_dict['code'])
except ValueError:
    raise ValidationError("Code not valid")     
```